CONTRIBUTING.md

```diff
@@ -27,9 +43,21 @@ Code Stuff
   - The only exception: a `switch` statement with all `case` statements being a single short statement is allowed to use the short brace-less form.
   - These two rules really mean that indent is governed by braces
  * Add an empty last line in all source files (GCC and GIT can complain otherwise)
- * All new public functions in all classes need documenting comments on what they do and what behavior they follow, use doxy-comments formatted as `/** Description */`. Do not use asterisks on additional lines in multi-line comments.
- * Use spaces after the comment markers: `// Comment` instead of `//Comment`
 
+Pre-commit hook
+---------
+When contributing, the code conventions above *must* be followed. Otherwise, the CI builds will automatically fail and your PR will not be merged until the non-conforming code is fixed. It is highly recommended to set up a pre-commit hook which will check your code style before every commit. Here is how to do that:
+
+ * Clone the repository as usual. 
+ * Go to your `<clone location>/.git/hooks` folder, create a text file named "pre-commit" there with the following contents:
+```
+#!/usr/sh
+src/CheckBasicStyle.lua 1>&2 -g
+```
+ * You need to give the newly created file an execute permission: `chmod +x .git/hooks/pre-commit`
```

Comment:
Not really Linux only, it should work on any Unix like OS.
On 7 May 2015 08:35, "Mattes D" <notifications@github.com> wrote:

> In CONTRIBUTING.md
> <https://github.com/mc-server/MCServer/pull/1957#discussion_r29830220>:
>
> >
> > +Pre-commit hook
> > +---------
> > +When contributing, the code conventions above *must* be followed. Otherwise, the CI builds will automatically fail and your PR will not be merged until the non-conforming code is fixed. It is highly recommended to set up a pre-commit hook which will check your code style before every commit. Here is how to do that:
> > +
> > + * Clone the repository as usual.
> > + * Go to your `<clone location>/.git/hooks` folder, create a text file named "pre-commit" there with the following contents:
> > +```
> > +#!/usr/sh
> > +src/CheckBasicStyle.lua 1>&2 -g
> > +```
> > + * You need to give the newly created file an execute permission: `chmod +x .git/hooks/pre-commit`
>
> This is Linux-only, perhaps mention that?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mc-server/MCServer/pull/1957/files#r29830220>.
>